### ISSUE TITLE
[Coil 3] Having compile time error: Could not find "io.ktor:ktor-io-cinterop-mutex". + Repro

### ISSUE THREAD

**Issue by rustamsmax** [coil-kt/coil#1968]
Having compile time error when building ios app

Xcode build output:
<img width="495" alt="image" src="https://github.com/coil-kt/coil/assets/36570577/0d06675a-93bb-425e-84f5-05e6fc371f20">

Android studio compileKotlinIosArm64 output:
![telegram-cloud-photo-size-2-5201859373402546818-y](https://github.com/coil-kt/coil/assets/36570577/ee1595ee-2e5e-4fd4-bdd7-6511a3e0cf06)

Here's a repro
https://github.com/rustamsmax/coil3-repro

I've opened an issue #1967  before but it was closed without investigation

**Comment by colinrtwhite** [comment]
Apologies - looks like I closed that ticket too quickly. I'm able to reproduce the issue locally with your sample.

**Comment by colinrtwhite** [comment]
I was able to work around this by using a newer version of Ktor - specifically the experimental wasm version. My hunch is the latest non-experimental versions of Ktor were published with a bug that occurred in Kotlin plugin 1.9.20 for native libraries. Hopefully it's fixed in the next `3.0.0` beta release or stable patch. Here's how to use the wasm build for now:

```kotlin
repositories {
    maven("https://maven.pkg.jetbrains.space/kotlin/p/wasm/experimental")
}

dependencies {
    implementation("io.ktor:ktor-client-core:3.0.0-wasm2")
}

**Comment by DRSchlaubi** [comment]
I can't use ktor 3.0 (for reasons), so downgrading to Ktor 2.3.6 seems to work, however I haven't fully tested it

**Comment by rustamsmax** [comment]
with latest alpha 01 release ios target was built successfully. seems wasm artifact related issue

**Comment by colinrtwhite** [comment]
@rustamsmax Looks like this was fixed with the latest `2.3.8`. Updating to Coil `3.0.0-alpha04`, which imports that version, fixes the repro sample.
